reports/tests.py

```diff
@@ -528,3 +533,110 @@ def _generate_invoice_date(year=datetime.now().year, month=1, this_month=False):
             date = datetime(year, month, 1)
 
         return timezone.make_aware(date)
+
+
+class ReportTaskTestCase(TestCase):
+    """
+    Tests for reports tasks execution
+    """
+
+    def test_send_trial_instances_report_failure(self):
+        """
+        Tests that if send_trial_instances_report fails because there are no matching instances
+        the recipient gets notified regardless.
+        """
+        send_trial_instances_report(recipients=['to@example.com'])
+        sent_mail = mail.outbox[0]
+        self.assertIn(
+            'Unable to generate a Trial Instances Report due to failure of `instance_statistics_csv`',
+            sent_mail.body
+        )
+        self.assertEqual(
+            [],
+            sent_mail.attachments
+        )
+        self.assertEqual(
+            ['to@example.com'],
+            sent_mail.to
+        )
+        self.assertEqual(len(mail.outbox), 1)
+
+    def test_send_trial_instances_report(self):
+        """
+        Tests that if send_trial_instances_report succeeds, the recipient receives
+        a correct report with only instances created last month.
+        """
+        date1 = "2020-01-01"
+        date2 = "2020-02-01"
+        date3 = "2020-03-01"
+
+        user1, _ = get_user_model().objects.get_or_create(
+            username='test1', email='test1@example.com')
+        user2, _ = get_user_model().objects.get_or_create(
+            username='test2', email='test2@example.com')
+        user3, _ = get_user_model().objects.get_or_create(
+            username='test3', email='test3@example.com')
+
+        with freeze_time(date1):
+            instance1 = OpenEdXInstanceFactory(
```

Comment:
@spokerman12, the repeated code below for creating multiple users can be refactored into a separate function or a factory function, no?

Reply:
Because it's implied that there's an ordering  user1 < user2 < user3, instance1 < instance2 < instance3 ... etc, because we're adding a singular feature I prefer the test to be explicit (or "don't make me think," as suggested in the handbook).

I'd add the factory function if we were adding more tests that could make use of it, but in this case I don't think it's necessary.